cmd/network/packet-capture.go

```diff
@@ -325,3 +363,138 @@ func copyFilesFromPacketCapturePods(o *packetCaptureOptions) error {
+func desiredPacketCapturePod(o *packetCaptureOptions, key types.NamespacedName) *corev1.Pod {
+	capturePod := &corev1.Pod{}
+	t := true
+	ls := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": key.Name,
+		},
+	}
+	capturePod.Name = key.Name
+	capturePod.Namespace = key.Namespace
+	capturePod.Labels = ls.MatchLabels
+	capturePod.Spec.NodeSelector = map[string]string{
+		o.nodeLabelKey: o.nodeLabelValue,
+	}
+	capturePod.Spec.Tolerations = []corev1.Toleration{
+		{
+			Effect:   "NoSchedule",
+			Key:      o.nodeLabelKey,
+			Operator: "Exists",
+		},
+	}
+	capturePod.Spec.Volumes = []corev1.Volume{
+		{
+			Name: "capture-output",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	capturePod.Spec.HostNetwork = true
+	capturePod.Spec.InitContainers = []corev1.Container{
+		{
+			Name:            "init-capture",
+			Image:           packetCaptureImage,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command:         []string{"/bin/bash", "-c", "tcpdump -G " + strconv.Itoa(o.duration) + " -W 1 -w /tmp/capture-output/capture.pcap -i vxlan_sys_4789 -nn -s0; sync"},
+			SecurityContext: &corev1.SecurityContext{Privileged: &t},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "capture-output",
+					MountPath: "/tmp/capture-output",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+	capturePod.Spec.Containers = []corev1.Container{
+		{
+			Name:            "copy",
+			Image:           packetCaptureImage,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			Command:         []string{"/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"},
+			SecurityContext: &corev1.SecurityContext{Privileged: &t},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "capture-output",
+					MountPath: "/tmp/capture-output",
+					ReadOnly:  false,
+				},
+			},
+		},
+	}
+	return capturePod
+}
+
+// hasPacketCapturePod returns the current daemonset
+func hasPacketCapturePod(o *packetCaptureOptions, key types.NamespacedName) (bool, error) {
+	capturePod := &corev1.Pod{}
+
+	if err := o.kubeCli.Get(context.TODO(), key, capturePod); err != nil {
+		if k8serr.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// deletePacketCapturePod creates the given daemonset resource
+func deletePacketCapturePod(o *packetCaptureOptions, capturePod *corev1.Pod) error {
+	if err := o.kubeCli.Delete(context.TODO(), capturePod); err != nil {
+		return fmt.Errorf("failed to delete Pod %s/%s: %v", capturePod.Namespace, capturePod.Name, err)
+	}
+	return nil
+}
+
+// ensurePacketCapturePod ensures the daemonset exists
+func ensurePacketCapturePod(o *packetCaptureOptions) (*corev1.Pod, error) {
+	key := types.NamespacedName{Name: o.name, Namespace: o.namespace}
+	desired := desiredPacketCapturePod(o, key)
+	havePod, err := hasPacketCapturePod(o, key)
+	if err != nil {
+		log.Fatalf("Error getting current Pod %v", err)
```

Comment:
Perhaps "Error checking for current Pod"?  As is, and without context, it might imply it failed to return a pod or something.

Reply:
Hey. I do not get this comment :/ could you elaborate a bit more?

Reply:
I think it's just semantic confusion. The wording of the error implies, at least to me, that you were trying to get and return a pod which, fair, that's what `kubeCli.Get()` is doing, but since having a received or not received a pod is true or false in the value of `havePod`, the error would mean there was more a problem just performing the operation.

Feel free to ignore this.  I might just be how I'm interpreting the error string. Either way it's not critical in the least.  Just a suggestion.